spec/abstractops.html

```diff
@@ -1510,7 +1510,9 @@ <h1>ToPositiveIntegerOrInfinity ( _argument_ )</h1>
   <emu-clause id="sec-temporal-preparetemporalfields" aoid="PrepareTemporalFields">
     <h1>PrepareTemporalFields ( _fields_, _fieldNames_, _requiredFields_ )</h1>
     <p>
-      The abstract operation PrepareTemporalFields reads the relevant properties of an Object _fields_; ensures all the required properties are present; and creates a new Object with all the properties converted to the correct type. It performs the following steps:
+      The abstract operation PrepareTemporalFields reads the relevant properties of an Object _fields_; ensures all the required properties are present; and creates a new Object with all the properties converted to the correct type.
+      The relevant property names are given by the List of Strings _fieldNames_, and the required property names are given by the List of Strings _requiredFields_.
+      It performs the following steps:
```

Comment:
so for a required field, "present and undefined" and "absent" will throw, but "present and null" is property-dependent?

Reply:
That seems in line with other places that use ToIntegerOrInfinity or ToString in JS — `era: null` is coerced to `era: "null"`, and `year: null` is coerced to `year: 0`

Reply:
`undefined` in those cases is coerced to `"undefined"` or `NaN` (probably `0` for ToIntegerOrInfinity) - i'm asking about the throw for undefined, not the null :-)

Reply:
I see, I thought you were confirming the throw for undefined and complaining about the lack of throwing for null. Yes, present but undefined does throw - steps 3a-b of PrepareTemporalFields.